source/extensions/filters/http/oauth/filter.cc

```diff
@@ -0,0 +1,475 @@
+#include "extensions/filters/http/oauth/filter.h"
+
+#include <algorithm>
+#include <cstdlib>
+#include <ctime>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "common/common/assert.h"
+#include "common/common/empty_string.h"
+#include "common/common/enum_to_int.h"
+#include "common/common/fmt.h"
+#include "common/common/hex.h"
+#include "common/common/matchers.h"
+#include "common/crypto/utility.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/headers.h"
+#include "common/http/message_impl.h"
+#include "common/http/utility.h"
+#include "common/protobuf/utility.h"
+
+#include "absl/strings/escaping.h"
+#include "absl/strings/match.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_replace.h"
+#include "absl/strings/str_split.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Oauth {
+
+// X-Forwarded Oauth headers
+const std::string Token{"token"};
```

Comment:
i think this and a bunch of the other strings here can just be `constexpr absl::string_view`s, and you could have the things that use this move over to using `absl::string_view` instead of a string const ref